ministryofjustice/laa-apply-for-criminal-legal-aid
# [CRIMAPP-827] task list update

## Description of change
Updates the task list in preparation for adding the partner details section 
Introduces the passporting benefit section separating the dwp check flow from the client details section 
Refactors nino form to only reset attributes if the has nino or nino attribute has changed (previously was being updated every time the form was saved)
Persists the confirm dwp check screen to allow passporting benefit check answers to be validated

## Link to relevant ticket

[CRIMAPP-844](https://dsdmoj.atlassian.net/browse/CRIMAPP-844)

## Notes for reviewer
We may want to move the benefit attributes to their own table e.g. `passporting_benefit`. Most attributes are currently in the `person` table whereas now the benefit routes have been refactored out of client details, they should either live in the `crime_application` table or their own table. Thoughts/opinions welcome

We may also want to consider persisting the response to `Check your client details` screen to make the passporting benefit answer validation checks more accurate. At the moment the answers validator is relying on logic around which attributes are present to determine which flow the user has gone through and therefore which attributes should be validated. This also means it relies on the attributes to be reset if a user was to change their previous answers (where applicable)

## Screenshots of changes (if applicable)

### Before changes:

### After changes:

### Tasklist updates: 

**No nino or benefit**

https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/c1a14679-b7c2-482f-88b6-c993cbe09759

**Under 18** 

https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/48ec6309-3151-4793-b6b4-f2a3df006bc4

**Appeal no changes**

https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/78dc438c-8d6c-4602-ae16-1de6d4e2a655

**No nino + has benefit**

https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/b285e084-1f93-49a4-ae72-43a200b44317

**Has nino + has benefit**
**Confirms dwp result**

https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/94830d38-351b-494f-964b-8c9d20d36369

**Does not confirm dwp result**

https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/fc7adb39-cb05-419d-b59a-5ecaed5dd08b

**Has nino + has no benefit**

https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/2b3f736b-69a9-445a-993b-09975b4e4854

Nino form changes: 


## How to manually test the feature
Test the benefit/dwp check flows [using this figma link](https://www.figma.com/file/thkvnDBbQbHlSqzt1TE3lk/Crime-Apply?type=design&node-id=12580-51215&mode=design&t=r4LlqaayZUNuO48s-0) as well as the tasklist behaviour for under 18 applications and appeal no changes applications 

[CRIMAPP-844]: https://dsdmoj.atlassian.net/browse/CRIMAPP-844?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
